Cubical/ZCohomology/GroupStructure.agda

```diff
@@ -406,10 +544,45 @@ coHomGr n A = coHom n A , coHomGrnA
 ×coHomGr : (n : ℕ) (A : Type ℓ) (B : Type ℓ') → Group
 ×coHomGr n A B = dirProd (coHomGr n A) (coHomGr n B)
 
+coHomGroup : (n : ℕ) (A : Type ℓ) → AbGroup {ℓ}
+fst (coHomGroup n A) = coHom n A
+AbGroupStr.0g (snd (coHomGroup n A)) = 0ₕ n
+AbGroupStr._+_ (snd (coHomGroup n A)) = _+ₕ_ {n = n}
+AbGroupStr.- snd (coHomGroup n A) = -ₕ_ {n = n}
+IsAbGroup.isGroup (AbGroupStr.isAbGroup (snd (coHomGroup n A))) = isGroup (snd (coHomGr n A))
+IsAbGroup.comm (AbGroupStr.isAbGroup (snd (coHomGroup n A))) = commₕ n
+
+-- Reduced cohomology group (direct def)
+
+coHomRedGrDir : (n : ℕ) (A : Pointed ℓ) → Group {ℓ}
+fst (coHomRedGrDir n A) = coHomRed n A
+0g (snd (coHomRedGrDir n A)) = 0ₕ∙ n
+GroupStr._+_ (snd (coHomRedGrDir n A)) = _+ₕ∙_ {n = n}
+- snd (coHomRedGrDir n A) = -ₕ∙_ {n = n}
+isGroup (snd (coHomRedGrDir n A)) = helper
+  where
+  abstract
+    helper : IsGroup (0ₕ∙ n) (_+ₕ∙_ {n = n}) (-ₕ∙_ {n = n})
+    helper = makeIsGroup § (assocₕ∙ n) (rUnitₕ∙ n) (lUnitₕ∙ n) (rCancelₕ∙ n) (lCancelₕ∙ n)
+
+coHomRedGroupDir : (n : ℕ) (A : Pointed ℓ) → AbGroup {ℓ}
```

Comment:
Shouldn't the name have "Ab" in it? 

Also, why do you have `coHomRedGrDir` at all when you can get it from this lemma by projection?

Reply:
Actually ignore my comment about "Ab", the name is fine